### PR TITLE
[test] Allow running tests explicitly

### PR DESCRIPTION
**Pull request by Grub4K**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Allows running `hatch test test/test_utils.py::TestUtil::test_sanitize_path` directly, so that you don't have to wait for gathering (takes long especially on Windows).

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] Core bug fix/improvement

</details>

### CODE REVIEW

File: devscripts/run_tests.py [hunk 1]
```diff
@@ -40,8 +39,10 @@ def run_tests(*tests, pattern=None, ci=False):
     elif run_download:
         arguments.extend(['-m', 'download'])
     else:
-        arguments.extend(
-            f'test/test_download.py::TestDownload::test_{test}' for test in tests)
+        for test in tests:
+            if '/' not in test and '::' not in test:
```
Comment (bashonly) [inline devscripts/run_tests.py:43]: woudn't checking for `/` be enough? do we want to allow passing a full test module?
```suggestion
            if '/' not in test:
```

Comment (Grub4K) [reply devscripts/run_tests.py:43]: Yes I think you are right, everything will have to use the full qualified name, I'll change it

File: devscripts/run_tests.py [hunk 2]
```diff
@@ -27,7 +27,6 @@ def parse_args():
 def run_tests(*tests, pattern=None, ci=False):
     run_core = 'core' in tests or (not pattern and not tests)
     run_download = 'download' in tests
```
Comment (bashonly) [inline devscripts/run_tests.py:29]: maybe these need to be more strict?

Comment (Grub4K) [reply devscripts/run_tests.py:29]: Imo no, if theres `core` in the commandline itll apply markers. Theres 0 reason to use both markers and specify an explicit test. Using markers will still collect all tests, which then doesnt benefit `-k` as well

**Review by bashonly — APPROVED** [review]

**Pull request merged**